src/audio.rs

```diff
@@ -107,18 +113,21 @@ impl AudioCallback for Audio {
             let samples = source.samples.read().unwrap();
             for x in out.iter_mut() {
                 // Move forward
-                source.position += 1;
-                if source.position >= samples.len() {
+                source.position += source.pitch;
+                let mut position = source.position as usize; // Truncates
```

Comment:
We could do some linear interpolatio here. I'm not sure how much you hear it though, so this might be "good enough".

Reply:
The time between two samples is around 0.02 ms at 48000 Hz so unless you load a sample with extremely low sample rate you shouldn't be hearing anything.